src/base/cg.rs

```diff
@@ -70,6 +71,17 @@ impl<N: RealField> Matrix3<N> {
     pub fn new_rotation(angle: N) -> Self {
         Rotation2::new(angle).to_homogeneous()
     }
+
+    /// Creates a new homogeneous matrix that applies a scaling factor for each dimension with respect to point.
+    ///
+    /// Can be used to implement "zoom_to" functionality.
+    #[inline]
+    pub fn new_nonuniform_scaling_wrt_point(scaling: Vector2<N>, pt: Point2<N>) -> Self {
```

Comment:
Arguments should be provided by-ref to be coherent with the remaining of the cg api.

```suggestion
    pub fn new_nonuniform_scaling_wrt_point(scaling: &Vector2<N>, pt: &Point2<N>) -> Self {
```

Reply:
Why does matrix3::new_rotation (and others) take the argument by value?

Reply:
You are right. This looks inconsistent. We might need to revisit this at some point in the future, and settle in one conversion.